### PR TITLE
Adding code snippet views for SELECT, PUT and DELETE

### PR DESCRIPTION
**Pull request by nanoscott**
- Adding an unstyled tab
- Adding code tabs to allow for multiple languages
- Adding Dig views to allow for different operations

### CODE REVIEW

**Comment by nanoscott** [comment]
I signed it!

SCOTT HEAD
Senior Software Developer
E scott@nanopay.net | M +1.647.921.9217 | P +1.416.900.1111
nanopay Corporation | 171 East Liberty Street, Suite 340
Toronto, ON M6K 3P6
CANADA
www.nanopay.net

File: src/foam/doc/CodeTab.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+foam.CLASS({
```
Comment (kgrgreer) [inline src/foam/doc/CodeTab.js]: Add copyright header for this and other files.

File: src/foam/doc/CodeTab.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  package: 'foam.doc',
+  name: 'CodeTab',
+  properties: [
+    {
+      name: 'title',
+    },
+    {
+      name: 'code',
+    },
```
Comment (kgrgreer) [inline src/foam/doc/CodeTab.js]: No trailing commas.

File: src/foam/doc/DigDeleteView.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  extends: 'foam.u2.View',
+  implements: [
+    'foam.mlang.Expressions',
+  ],
+  imports: [
+    'appConfig',
+    'user',
+  ],
+  requires: [
+    'foam.doc.CodeTabs',
```
Comment (kgrgreer) [inline src/foam/doc/DigDeleteView.js:17]: sort requires

File: src/foam/doc/DigDeleteView.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
+foam.CLASS({
+  package: 'foam.doc',
+  name: 'DigDeleteView',
+  extends: 'foam.u2.View',
+  implements: [
```
Comment (kgrgreer) [inline src/foam/doc/DigDeleteView.js]: Add documentation:

File: src/foam/doc/DigPutView.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
+foam.CLASS({
+  package: 'foam.doc',
+  name: 'DigPutView',
+  extends: 'foam.u2.View',
+  implements: [
```
Comment (kgrgreer) [inline src/foam/doc/DigPutView.js]: documentation:

File: src/foam/doc/DigPutView.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    'foam.mlang.Expressions',
+  ],
+  imports: [
+    'appConfig',
+    'user',
+  ],
+  requires: [
+    'foam.doc.CodeTabs',
```
Comment (kgrgreer) [inline src/foam/doc/DigPutView.js:17]: sort

File: src/foam/doc/CodeTab.js (hunk 1 above)
Comment (mcarcaso) [inline src/foam/doc/CodeTab.js]: This model isn't being used. You can remove it and any requires of it and nothing should change.

File: src/foam/u2/UnstyledTabs.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
+/**
```
Comment (mcarcaso) [inline src/foam/u2/UnstyledTabs.js:1]: Use this header instead:
```
/**
 * @license
 * Copyright 2018 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */
```

File: src/files.js [hunk 8]
```diff
@@ -491,6 +492,8 @@ FOAM_FILES([
   { name: "foam/net/auth/TokenBearerHTTPRequest" },
   { name: "foam/net/auth/AuthAgent" },
   { name: "foam/doc/ClassLink", flags: ['debug', 'web'] },
+  { name: "foam/doc/CodeTab", flags: ['debug', 'web'] },
```
Comment (mcarcaso) [inline src/files.js]: Be sure to remove these before committing. The only thing you should need to add to files.js is the UnstyledTabs

File: src/foam/doc/DigSelectView.js [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+    {
+      name: 'url',
+      expression: function(appConfig) {
+        if ( appConfig ) return appConfig.url;
+      }
+    },
+    {
+      name: 'samplequery',
+      experssion: function(appConfig) {
+        if ( appConfig ) return appConfig.q;
```
Comment (mcarcaso) [inline src/foam/doc/DigSelectView.js]: appConfig.q isn't a thing. This is the AppConfig model:
https://github.com/foam-framework/foam2/blob/master/src/foam/nanos/app/AppConfig.js

File: src/foam/doc/DigSelectView.js [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+    'foam.doc.PropertyAxiom',
+    'foam.doc.CodeTab',
+    'foam.flow.PromiseSlot',
+    'foam.u2.DetailView',
+    'foam.nanos.dig.DIG',
+  ],
+  properties: [
+    {
+      class: 'StringArray',
+      name: 'digProperties',
```
Comment (mcarcaso) [inline src/foam/doc/DigSelectView.js]: You don't appear to be using this at all. Remove it from here and remove the reference to it in the slot below.

File: src/foam/doc/DigSelectView.js [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+    'foam.mlang.Expressions',
+  ],
+  imports: [
+    'appConfig',
+    'user',
+  ],
+  requires: [
+    'foam.doc.CodeTabs',
+    'foam.u2.Tab',
+    'foam.doc.dao.AxiomDAO',
```
Comment (mcarcaso) [inline src/foam/doc/DigSelectView.js]: Remove any unused requires. AxiomDAO, PropertyAxiom, CodeTab, DetailView, PromiseSlot

Comment (nanoscott) [reply src/foam/doc/DigSelectView.js]: I'm not exactly sure how I know if something is used or not .. there is no clear referencing mechanism, beside checking where fields come through text search. I'll just play around with removing things and if they break I'll add them back I guess.

Comment (nanoscott) [reply src/foam/doc/DigSelectView.js]: was pretty simple after all .. just removed everything that wasn't referenced in a 'self.' reference.

File: src/foam/doc/DigDeleteView.js [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+ foam.CLASS({
+  package: 'foam.doc',
+  name: 'DigDeleteView',
+  extends: 'foam.u2.View',
+  documentation: 'View used to show snippets of deletion API calls.',
+  implements: [
+    'foam.mlang.Expressions',
```
Comment (mcarcaso) [inline src/foam/doc/DigDeleteView.js]: I don't think you need this implements. This adds functions for creating expressions easily but I don't think you're using any here.

Comment (nanoscott) [reply src/foam/doc/DigDeleteView.js]: @mcarcaso .. interesting. how can you see either the functionality added or the source class that adds these methods? ... found it in mlang.js

Comment (mcarcaso) [reply src/foam/doc/DigDeleteView.js]: If you look at the foam.mlang.Expressions model you'll see what it adds. It lives in src/foam/mlang/mlang.js . It's a bit manual of a process to see what's added I'm afraid.

Comment (nanoscott) [reply src/foam/doc/DigDeleteView.js]: found it, thanks mike .. i've remove it from the views and everything works as expected.

**Review by kgrgreer — APPROVED** [review]

**Pull request merged**